app/models/story.rb

```diff
@@ -1020,7 +1023,7 @@ def fetched_attributes
 
   def self.title_maximum_length
     validators_on(:title)
-      .sole { |v| v.is_a? ActiveRecord::Validations::LengthValidator }
+      .find { |v| v.is_a? ActiveRecord::Validations::LengthValidator }
```

Comment:
Why doesn't `sole` work here anymore? Does ActiveRecord implement the `presence` validator as a second `LengthValidator`?

Reply:
`sole` expects 1 but there are now two validations on `title`, the `length` and now the `presence`. Sole checks the length of the enumerable before it applies the block.

Reply:
Oh, thank you. My incorrect mental model was that it was like `select`, returning a single object for which the block returned non-`nil` and raising an exception if there were 0 or more than one items. ([Array#sole doc](https://edgeapi.rubyonrails.org/classes/Enumerable.html#method-i-sole))

The block I wrote is basically a no-op. It sounds like the right code would be `.find { |v| v.is_a? ActiveRecord::Validations::LengthValidator }.sole.options[:maxiumum]` - does that look right to you?

Reply:
It's a bit weird. I don't know why it takes a block. Selecting the items and then asserting that there is only one makes sense. Thinking about it a bit more maybe `select` would be better than `find` but the spec passes...

```
irb(main):001> [1,2,3].sole {_1.even?}
/Users/henry/.gem/ruby/3.3.0/gems/activesupport-7.1.3.2/lib/active_support/core_ext/enumerable.rb:210:in `sole': multiple items found (Enumerable::SoleItemExpectedError)
irb(main):002> [1,2,3].select {_1.even?}
=> [2]
irb(main):003> [1,2,3].select {_1.even?}.sole
=> 2
irb(main):004> 
```

Reply:
I get it, it doesn't take a block, but Ruby functions accept a block by default, and the language doesn't enforce they'll get called because there are lots of useful functions that only conditionally call their blocks.

```
irb(main):001> def foo
irb(main):002*   puts 'foo'
irb(main):003*   end
=> :foo
irb(main):004> foo { puts 'bar' }
foo
=> nil
```

```
irb(main):001> [1,2,3].sole {puts 'bar'}
[path elided]/enumerable.rb:210:in `sole': multiple items found (Enumerable::SoleItemExpectedError)
```
Note that neither example prints `bar`.